shepherd/utils/logging_config.py

```diff
@@ -0,0 +1,116 @@
+
+
+class LoggerMaxInfoFilter(logging.Filter):
+    """
+    Logger filter, which allows logs only with level INFO or lower.
+    """
+    def filter(self, record: logging.LogRecord) -> bool:
+        return record.levelno <= logging.INFO
+
+
+def setup_logging_dict_config(logging_directory: str):
+    logging_config = {
+        "version": 1,
+        "disable_existing_loggers": True,
+        "filters": {
+            "max_info_filter": {
+                "()": LoggerMaxInfoFilter
+            }
+        },
+        "formatters": {
+            "terminal": {
+                "()": BaseFormatter
+            },
+            "file": {
+                "()": BaseFormatter
+            }
+        },
+        "handlers": {
+            "console_info": {
+                "class": "logging.StreamHandler",
+                "filters": ["max_info_filter"],
+                "level": "NOTSET",
+                "formatter": 'terminal',
+                "stream": "ext://sys.stdout"
+            },
+            "console_error": {
+                "class": "logging.StreamHandler",
+                "level": "WARNING",
+                "formatter": 'terminal',
+                "stream": "ext://sys.stderr"
+            },
+            "info_file": {
+                "class": "logging.handlers.RotatingFileHandler",
+                "level": "INFO",
+                "formatter": "file",
+                "filename": f"{logging_directory}/info_shepherd.log",
+                "maxBytes": 5000000,
+                "backupCount": 5,
+                "encoding": "utf-8"
+            },
+            "error_file": {
+                "class": "logging.handlers.RotatingFileHandler",
+                "level": "ERROR",
+                "formatter": "file",
+                "filename": f"{logging_directory}/error_shepherd.log",
+                "maxBytes": 5000000,
+                "backupCount": 5,
+                "encoding": "utf-8"
+            }
+        },
+        "loggers": {
+            "": {
+                # root logger
+                "level": "NOTSET",
+                "handlers": [
+                    "console_info", "console_error", "info_file", "error_file"
+                ],
+            },
+            "aiohttp.server": {
+                "level": "WARNING",
+                "handlers": [
+                    "console_info", "console_error", "info_file", "error_file"
+                ]
+            }
+        }
+    }
+
+    dictConfig(config=logging_config)
+
+
+def setup_logging(logging_level: int = logging.INFO,
+                  logging_directory: str = '../logs'):
+    Path(logging_directory).mkdir(parents=True, exist_ok=True)
+    setup_logging_dict_config(logging_directory)
+
+    logging.getLogger("urllib3").setLevel(logging.WARNING)  # switch off unnecessary logs
+
+    if isinstance(logging_level, int):
+        logging.disable(level=logging_level - 10)  # set minimum logging level for all loggers
```

Comment:
co je tohleto? Proc -10? proc kdyz pises ze nastavis minimum level to nenastavis na konkretni hodnotu?

Reply:
Protože metoda logging.disable(level: int = ...) vypne všechny logy s level "level" a níž. To znamená, že pokud chceme vypnout logy pod určitým "level", musíme na vstupní argument "level" u funkce disable poslat level o jeden níž. Logging levels v Pythonu jsou definovány jako int jdoucí po 10

Reply:
Tak to můžeš dát na nejnižší ne? Level. None?

Reply:
Nějak nechápu tohle se děje automaticky že logy níž než nastavený level se vypnou ne? Proč to musíš ty dělat?

Reply:
Už jsem to našel. Předtím jsme to nastavovali v basicConfig (teď v dictConfig). Navíc pokud jsme nastavovali logging level, tak ten působil jenom pro root logger, když teď bych chtěl to mít pro všechny loggery, které máme a další, které by mohli mít v budoucnosti. Tak jsem to nastavil pro všechny handlery, které máme a asi teď to má být míň matoucí, než s tím disable